packages/provider/src/transports/wallet-request-handler.ts

```diff
@@ -471,12 +474,14 @@ export class WalletRequestHandler implements ExternalProvider, JsonRpcHandler, P
           // and would have prompted the user upon signing.
 
           // https://eth.wiki/json-rpc/API#eth_sendRawTransaction
-          if (commons.transaction.isSignedTransactionBundle(request.params![0])) {
-            const txChainId = BigNumber.from(request.params![0].chainId).toNumber()
-            const tx = await account.relayer(txChainId)!.relay(request.params![0])
+          const transaction = request.params![0]
+          if (commons.transaction.isSignedTransactionBundle(transaction)) {
+            validateTransactionRequest(account.address, transaction.transactions)
```

Comment:
I don't think is flow is even properly supported, but just in case this should happen on the sign, not on the send. After the transaction is signed is already too late.